File: Plugins/URLSanitize.py
from Plugins import Plugin

import re

class URLSanitize(Plugin.Plugin):
  def __init__(self, url: str = None, name: str = 'URLSanitize'):
    super().__init__(name)
    self._url = url

  def _sanitize(self, url) -> str:
    x = re.sub(r"\.", "[.]", url)
    x = re.sub("http://", "hxxp://", x)
    x = re.sub("https://", "hxxps://", x)
    return x

  def run(self):
    print("\n --------------------------------- ")
    print(" U R L   S A N I T I S E   T O O L ")
    print(" --------------------------------- ")
    if self._url == None:
      self._url = input('Enter URL to sanitize: ').strip()

    print("\n" + self._sanitize(self._url))

File: src/plugins/url_sanitize.rs
use super::Plugin;
use serde_json;

use crate::utils::get_input;

pub struct URLSanitize {}

impl Plugin for URLSanitize {
  const NAME: &'static str = "URL Sanitize";
  fn new() -> Self {Self {}}

  fn perform_lookup(&self, value: &str) -> Result<serde_json::Value, String> {
    let sanitized_url = Self::sanitize_url(value);
    Ok(serde_json::json!({"result": "sanitized", "value": sanitized_url}))
  }

  fn run(&self) {
    println!("\n --------------------------------- ");
    println!(" U R L   S A N I T I Z E   T O O L ");
    println!(" --------------------------------- ");

    let input_url = get_input("Enter a URL to sanitize: ");
    match self.perform_lookup(&input_url) {
      Ok(result) => println!("\n{}", result),
      Err(error) => eprintln!("Error: {}", error),
    }
  }

  fn get_name(&self) -> &str {
    return Self::NAME;
  }
}

impl URLSanitize {
  fn sanitize_url(url: &str) -> String {
    let sanitized_url = url.replace(".", "[.]").replace("http://", "hxxp://").replace("https://", "hxxps://");
    sanitized_url
  }
}